test/modules/EdgeHubRestartTester/src/DirectMethodEdgeHubConnector.cs

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace EdgeHubRestartTester
+{
+    using System;
+    using System.Net;
+    using System.Net.Sockets;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Azure.Devices.Client.Exceptions;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil.TestResults;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    class DirectMethodEdgeHubConnector : IEdgeHubConnector
+    {
+        long directMethodCount = 0;
+
+        ModuleClient DmModuleClient { get; }
+        Guid BatchId { get; }
+        DateTime RunExpirationTime { get; }
+        CancellationToken CancellationToken { get; }
+        DateTime EdgeHubRestartedTime { get; }
+        HttpStatusCode EdgeHubRestartStatusCode { get; }
+        uint RestartSequenceNumber { get; }
+        ILogger Logger { get; }
+
+        public DirectMethodEdgeHubConnector(
+            ModuleClient dmModuleClient,
+            Guid batchId,
+            DateTime runExpirationTime,
+            CancellationToken cancellationToken,
+            DateTime edgeHubRestartedTime,
+            HttpStatusCode edgeHubRestartStatusCode,
+            uint restartSequenceNumber,
+            ILogger logger)
+        {
+            this.DmModuleClient = Preconditions.CheckNotNull(dmModuleClient, nameof(dmModuleClient));
+            this.BatchId = batchId;
+            this.RunExpirationTime = runExpirationTime;
+            this.CancellationToken = Preconditions.CheckNotNull(cancellationToken, nameof(cancellationToken));
```

Comment:
remove it; not applicable for struct.  And normally this argument should be the last one and use default keyword to make it optional

Reply:
I'm not making this optional. I want to make sure that the tasks are properly cancelled.